### PR TITLE
NoEmptyCommentFixer - Only remove complete empty comment blocks

### PR DESCRIPTION
**Pull request by SpacePossum**
master version of;
@see https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/1941

I've changed the spec a bit to only complete empty comment blocks.
Started on master because of the comment with line break transformers are helping a lot...

### CODE REVIEW

File: src/Fixer/Comment/NoEmptyCommentFixer.php [hunk 1]
```diff
@@ -20,22 +20,31 @@
...
+                continue;
+            }
+
+            list($blockStart, $index, $isEmpty) = $this->getCommentBlock($tokens, $index);
+            if (false === $isEmpty) {
+                continue;
+            }
+
+            if ($tokens[$index]->isWhitespace() && false !== strpos($tokens[$index]->getContent(), "\n")) {
+                --$index;
```
Comment (keradus) [inline src/Fixer/Comment/NoEmptyCommentFixer.php]: why we are going backward once?

File: src/Fixer/Comment/NoEmptyCommentFixer.php [hunk 2]
```diff
@@ -58,34 +67,105 @@ public function getPriority()
     }
 
     /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
```
Comment (keradus) [inline src/Fixer/Comment/NoEmptyCommentFixer.php]: should not be moved (as, alphabetic order inside methods with same visibility)

File: src/Fixer/Comment/NoEmptyCommentFixer.php [hunk 3]
```diff
@@ -58,34 +67,105 @@ public function getPriority()
...
+                // single line comment starting with '#'
+                return 1 === preg_match('|^#\s*$|', $content);
+            case self::TYPE_SLASH_ASTERISK:
+                // comment starting with '/*' and ending with '*/' (but not a PHPDoc)
+                return 1 === preg_match('|^/\*\s*\*/$|', $content);
+            case self::TYPE_DOUBLE_SLASH:
+                // single line comment starting with '//'
+                return 1 === preg_match('|^//\s*$|', $content);
+            default:
+                throw new \UnexpectedValueException('Comment type detected failed.');
```
Comment (keradus) [inline src/Fixer/Comment/NoEmptyCommentFixer.php]: could never happen, as getCommentType return only 3 values (if it cannot detect it returns `//`)

**Comment by keradus** [comment]
Awesome work ! Thank you @SpacePossum 
Please, could you rebase  and see my tiny comment ?

File: src/Fixer/Comment/NoEmptyCommentFixer.php (hunk 2 above)
Comment (SpacePossum) [inline src/Fixer/Comment/NoEmptyCommentFixer.php]: but the methods same visibility are now in alphabetic order, you want `isCandidate` always above `fix` method ?

File: src/Fixer/Comment/NoEmptyCommentFixer.php (hunk 1 above)
Comment (SpacePossum) [inline src/Fixer/Comment/NoEmptyCommentFixer.php]: not needed indeed, relic from 1.x branch, removed.

File: src/Fixer/Comment/NoEmptyCommentFixer.php (hunk 2 above)
Comment (keradus) [inline src/Fixer/Comment/NoEmptyCommentFixer.php]: my bad, I ignored `is` prefix. please ignore me ;)

**Comment by keradus** [comment]
Thank you @SpacePossum.

**Comment by SpacePossum** [comment]
@keradus this fix should be backported to 1.x, however if I cannot do this in time for 1.12 release maybe we should remove the fixer for the 1.x line.

**Comment by GrahamCampbell** [comment]
I don't think we really need to backport. Just document the fixer behaviour has changed in the upgrading guide.

**Comment by keradus** [comment]
I don't see value for backporting this one as well, as it's sth in the middle of bugfix and specification update.
2.x in general change way of dealing with comments, so I don't see big need of describing this particular comment fixer change

**Comment by SpacePossum** [comment]
very well, saves a lot of coding ^-^ thanks! :+1:

**Pull request closed without merging**